src/test/kotlin/io/github/mojira/arisa/modules/ReopenAwaitingModuleTest.kt

```diff
@@ -500,12 +652,67 @@ class ReopenAwaitingModuleTest : StringSpec({
             updated = updated,
             reporter = REPORTER,
             comments = listOf(comment),
-            changeLog = listOf(AWAITING_RESOLVE, changeLog)
+            changeLog = listOf(AWAITING_RESOLVE, changeLog),
+            reopen = { hasReopened = true; Unit.right() },
+            addComment = { hasCommented = true; Unit.right() }
+        )
+
+        val result = MODULE(issue, TEN_SECONDS_AGO)
+
+        result.shouldBeRight(ModuleResponse)
+        hasReopened shouldBe true
+        hasCommented shouldBe false
+    }
+
+    "should comment the message when there is a keep AR tag" {
+        var hasReopened = false
+        var hasCommented = false
+
+        val updated = RIGHT_NOW.plusSeconds(3)
+        val tagComment = getComment(
+            body = "MEQS_KEEP_AR",
+            visibilityType = "group",
+            visibilityValue = "staff"
+        )
+        val normalComment = getComment()
+        val issue = mockIssue(
+            resolution = "Awaiting Response",
+            updated = updated,
+            reporter = REPORTER,
+            comments = listOf(tagComment, normalComment),
+            changeLog = listOf(AWAITING_RESOLVE),
+            reopen = { hasReopened = true; Unit.right() },
+            addComment = { hasCommented = true; Unit.right() }
+        )
+
+        val result = MODULE(issue, TEN_SECONDS_AGO)
+
+        result.shouldBeRight(ModuleResponse)
+        hasReopened shouldBe false
+        hasCommented shouldBe true
+    }
+
+    "should comment the message when someone answered after the soft AR period" {
+        var hasReopened = false
+        var hasCommented = false
+
+        val updated = RIGHT_NOW.plusSeconds(3)
+        val comment = getComment()
+        val issue = mockIssue(
+            resolution = "Awaiting Response",
+            updated = updated,
+            reporter = REPORTER,
+            comments = listOf(comment),
+            changeLog = listOf(OLD_AWAITING_RESOLVE),
+            reopen = { hasReopened = true; Unit.right() },
+            addComment = { hasCommented = true; Unit.right() }
         )
 
         val result = MODULE(issue, TEN_SECONDS_AGO)
 
         result.shouldBeRight(ModuleResponse)
+        hasReopened shouldBe false
+        hasCommented shouldBe true
     }
 
     "should return FailedModuleResponse with all exceptions when reopening fails" {
```

Comment:
I think the Failed tests are missing for when it fails to comment.

Reply:
We won't need them after #264 anyways

Reply:
Yeah this was posted slightly before that pr, or at least before I became aware of it